packages/runtime/datastore/src/fluidHandle.ts

```diff
@@ -36,40 +49,41 @@ export class FluidObjectHandle<T extends FluidObject = IFluidObject> implements
         public readonly routeContext: IFluidHandleContext,
     ) {
         this.absolutePath = generateHandleContextPath(path, this.routeContext);
+        if (this.isAttached) {
+            this._visible = true;
+        }
     }
 
     public async get(): Promise<any> {
         return this.value;
     }
 
     public attachGraph(): void {
-        // If this handle is already in attaching state in the graph or attached, no need to attach again.
-        if (this.graphAttachState !== AttachState.Detached) {
+        if (this.visible) {
             return;
         }
-        this.graphAttachState = AttachState.Attaching;
-        if (this.bound !== undefined) {
-            for (const handle of this.bound) {
+
+        this._visible = true;
+        if (this.boundHandles !== undefined) {
+            for (const handle of this.boundHandles) {
                 handle.attachGraph();
             }
 
-            this.bound = undefined;
+            this.boundHandles = undefined;
```

Comment:
why would we want to unbind all the bound handles?

Reply:
We already called attachGraph() on these handles. We are just emptying the boundHandles set now. We could explicitly clear the set but I don't think it matters.

Reply:
Wouldn't we want to keep the boundHandles stored, so when we unbind, we unbind the boundHandles?

Reply:
We don't have the concept of unbind today. boundHandles is basically a temporary list that contains handles that are bound to the data store when it is not visible. We use this list to actually bind these handles when we become visible.
So, it isn't really a list of all handles that are bound to this data store. The name is misleading and so are the terms attach and bind. That's why I am trying to move to just visible.
Maybe we can call this `pendingHandlesToBind` or `pendingHandlesToMakeVisible`?

Reply:
That would work, `pendingHandlesToMakeVisible`, since that's the method we're calling